packages/gatsby-dev-cli/src/local-npm-registry/install-packages.js

```diff
@@ -29,7 +29,49 @@ const installPackages = async ({
       { stdio: `pipe` },
     ])
 
-    const workspacesLayout = JSON.parse(JSON.parse(stdout).data)
+    let workspacesLayout
+    try {
+      workspacesLayout = JSON.parse(JSON.parse(stdout).data)
+    } catch (e) {
+      /*
+      Yarn 1.22 doesn't output pure json - it has leading and trailing text:
```

Comment:
:(
cc @arcanis

Reply:
- Use the `--silent` flag to remove the header / footer from the output
- I think in 1.x the `--json` flag will wrap the output into a "log" property which you probably don't want

In result, I think the command should be `yarn workspaces info --silent` in 1.x, and `yarn workspaces list --json -v` in 2.x.

Reply:
> Use the --silent flag to remove the header / footer from the output

Doesn't seem to work in 1.22 - in fact neither `silent` nor `json` really do - using `yarn workspaces info`, `yarn workspaces info --json`, `yarn workspaces info --silent`, `yarn workspaces info --json --silent` all output same thing:

```
$ yarn workspaces info --silent
yarn workspaces v1.22.0
warning package.json: No license field
{
  "z": {
    "location": "z",
    "workspaceDependencies": [],
    "mismatchedWorkspaceDependencies": []
  },
  "y": {
    "location": "y",
    "workspaceDependencies": [],
    "mismatchedWorkspaceDependencies": []
  }
}
Done in 0.20s.
```

> I think in 1.x the --json flag will wrap the output into a "log" property which you probably don't want

It does in <1.22 - it has `{ type: "log", data: "stringified-json" }` (hence grabbing `.data` in `JSON.parse(stdout).data`), it doesnt in >=1.22 (or at least in 1.22.0).

In any case - this is really not looking to make this 100% reliable - I do want to explore using Yarn 2 with `yarn link` that uses portals protocol - like you did for "e2e pnp tests" (just `gatsby-dev-cli` is always on backlog and is never firest fire to work on, so it's just matter of not being able to spend more time on it)

Reply:
> It does in <1.22 - it has `{ type: "log", data: "stringified-json" }` (hence grabbing `.data` in `JSON.parse(stdout).data`), it doesnt in >=1.22 (or at least in 1.22.0).

You're right, seems like a regression in the CLI parsing ... 😢

Will be good enough then 👍